kafka-rest/src/test/java/io/confluent/kafkarest/testing/KafkaRestFixture.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.testing;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.kafkarest.KafkaRestApplication;
+import io.confluent.kafkarest.KafkaRestConfig;
+import io.confluent.rest.RestConfig;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Properties;
+import javax.annotation.Nullable;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import org.eclipse.jetty.server.Server;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+public final class KafkaRestFixture implements BeforeEachCallback, AfterEachCallback {
+
+  @Nullable private final SslFixture certificates;
+  private final HashMap<String, String> configs;
+  private final KafkaClusterFixture kafkaCluster;
+  @Nullable private final String kafkaPassword;
+  @Nullable private final String kafkaUser;
+  @Nullable private final String keyName;
+  @Nullable private final SchemaRegistryFixture schemaRegistry;
+
+  @Nullable
+  private URI baseUri;
+  @Nullable
+  private KafkaRestApplication application;
+  @Nullable
+  private Server server;
```

Comment:
super-nit: Is the idea here to keep `@Nullable` on a different line so that readers will notice that these fields are non-`final`? I'd suggest keeping everything on the same line, just delimiting `final` and non-`final` fields with a blank line, like you've done already.

Reply:
Done.